services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainMetrics.java

```diff
@@ -181,17 +181,16 @@ public BeaconChainMetrics(
             "previous_epoch_total_weight",
             "Total effective balance of all active validators in the previous epoch");
 
-    final String versionLabel =
-        VersionProvider.IMPLEMENTATION_VERSION
-            .replaceAll("[\\.\\+]", "_")
-            .replaceAll("[^0-9A-Za-z_]", "");
+    final String version =
+        VersionProvider.IMPLEMENTATION_VERSION.replaceAll("^v", "").replaceAll("\\+.*$", "");
```

Comment:
I can try again, but prometheus wasn't on board with having labels that had a period in them, which was why the version had to contain '_'.
https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

labels can contain `[a-zA-Z_][a-zA-Z0-9_]*`, and labels starting with `_` are reserved for internal use...

It looks like the intent above is to make '22.2.0' a label, which was what i started trying to do, but it's not going to allow it because '.' isn't allowed in a label

The implementation version also had the '+' which needed stripping... so the initial versionLabel above is basically the prometheus friendly version of our implementation_version for a label.

Reply:
I might actually just check this out and see what happens when it runs...

Reply:
I think you're confusing label names and label values.  This should create a metric like:

```
beacon_teku_version{version="21.2.0",release="v21.2.0+45-ge13fd6e"} 1
```

The 'version' and 'release' label names are constant, their values are the current version and release, respectively.

Reply:
ok i completely retract my comment given that it's working!